coderd/database/queries/workspaces.sql

```diff
@@ -161,17 +178,41 @@ WHERE
 			name ILIKE '%' || @name || '%'
 		ELSE true
 	END
+	-- Filter by agent status
+	-- has-agent: is only applicable for workspaces in "start" transition. Stopped and deleted workspaces don't have agents.
+	AND CASE
+		WHEN @has_agent :: text != '' THEN
+			latest_build.transition = 'start'::workspace_transition
+			AND CASE
+				WHEN @has_agent = 'timeout' THEN
+					latest_build.first_connected_at IS NULL AND (latest_build.created_at + latest_build.connection_timeout_seconds * INTERVAL '1 second' < NOW())
+				WHEN @has_agent = 'connecting' THEN
+					latest_build.first_connected_at IS NULL
```

Comment:
This could also return true for an agent that has timed out?

Reply:
I guess that this is also possible. I included the extra condition.

Reply:
See above.